lms/djangoapps/shoppingcart/views.py

```diff
@@ -1022,3 +1031,118 @@ def csv_report(request):
 
     else:
         return HttpResponseBadRequest("HTTP Method Not Supported")
+
+
+class PurchaseView(APIView):
+    """Purchase Seats and Certificates for a Course. """
+
+    authentication_classes = OAuth2AuthenticationAllowInactiveUser, SessionAuthenticationAllowInactiveUser
+    throttle_classes = EnrollmentUserThrottle,
+
+    def post(self, request):
+        """Purchase Endpoint for calling the E-Commerce Service.
+
+        Call the E-Commerce Purchase application to complete and order and fulfill the associated enrollment.
+        This endpoint will short-circuit the E-Commerce Service if no SKU is found for the requested Course and
+        Certificate Type.
+
+        Use Cases
+            1. Purchase a Seat and associated Certificate for a Course.
+            2. Enroll the currently logged in user in a course. Currently you can use this command only to enroll
+                the user in "honor" mode. If honor mode is not supported for the course, the request fails and returns
+                the available modes.
+
+        Example Requests
+            POST /purchase{"course_details":{"course_id": "edX/DemoX/Demo_Course"}}
+
+        Post Parameters
+            * course details: A collection that contains:
+                * course_id: The unique identifier for the course.
+
+        """
+        user = request.user.username
+        if 'course_details' not in request.DATA or 'course_id' not in request.DATA['course_details']:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={"message": u"Course ID must be specified to purchase a seat."}
```

Comment:
Our REST API guidelines suggest providing a developer message and an i18n'd user-facing message in case of failure. I think that might be useful here.